### ISSUE TITLE
[Suggestion] Beam Function Changes

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#3098]
Though weapons are still in a very early state, I’ve noticed a bit of a fundamental issue with the beam weapons me and daOyster found during a fighter dog fight. We noticed that the beam weapon became our primary and most effective weapon even though we had other lasers. The beam doesn't scatter and lasts for a few seconds compared the infinantly more inaccurate lasers. If what I assume is the plan for weapons, lasers having a low rate of fire but more powerful and kinetics having a high ROF but less powerful, the beam would break the model by being both powerful and have an infinite ROF. Unless you want to nerf the damage into the ground, there needs to be a fundamental change. One that has “beams” shift away from the best fighter weapon but still effective on capital ships. And no, shortening the range won't help. Fighters are always really close to each other and I have a feeling larger ships will be as well.

Beams are currently “Fire -> Cool Down -> Fire” making it very easy to hit an opponent without the chance for them to evade. With lasers, you have to consider spread and leading your target. With beams you click and hold then wait a few seconds. 
A solution to this would be “Charge Up -> Fire -> Cool Down -> Charge up” see the gifs below. The player would have to hold and aim before they are able to fire making the odds of hitting lower thus being more balanced. Still an effective weapon, but not a copy or replacement for lasers and the eventual kinetics.

![Laser](https://user-images.githubusercontent.com/36980446/96977901-91fe9380-14d2-11eb-93e4-461050411093.gif)
![ezgif com-resize](https://user-images.githubusercontent.com/36980446/96979251-e48c7f80-14d3-11eb-98b7-6f2baf01dd5b.gif)
Imagine if the player just pressed the mouse without needing to charge.

Now just a cool down would be enough to help fundamentally balance beams with the other weapons though the specific damage/fire length values would still need to be tweaked, but I think there is a better option: a full spartan laser. A single short burst rather than a long lasting beam. Currently with the beam, you can track the enemy as fast as your mouse moves giving you a good change of hitting the enemy if they are anywhere in your sight though how long the weapon makes contact is variable. Instead with a single burst, the player has a single shot to hit before the weapon needs to cool and charge again. This would make the weapon functionally different adding to the variety. Capital ships would still use them as a heavy hitting weapon while fighters can still have the beam but it would be more of a secondary fighter "finisher" or anti-capital ship weapon.

Long Range: Laser, Kinetics 
Medium Range: Beam
Short Range: Flame Thrower

Laser: Medium-high damage, low ROF, high projectile speed
Kinetics: Low-medium damage, high ROF, medium projectile speed
Beam: High damage, single shot, infinite projectile speed
Flame Thrower: Low but constant damage, constant stream, low projectile speed

Missile: Variable depending on modules

(I’m not sure if this is your plan as it is based off what I was saying, but knowing how the different weapons would function and their differences would be nice for future reference)

**Comment by tsunamayo** [comment]
@ZachZent thanks for the detailed suggestion as usual.
One thing to keep in mind on your test case:
1 - it was a dogfight (small and rapid fighter vs small and rapid
2 - you are shooting and aiming in manual from what I understand.

Now a lot of your feedback and hence the proposed solutions dont translate well to what will be the "meat" of the game, aka med-big ship fight using turrets.
I just want to get to that meat part first before I make further deep structural change to the beam, as I need to balance for all use cases. 
Single burst could work indeed, but I think we loose the aspect of what feels good about beam. 
Cheers

**Comment by ZachZent** [comment]
Fair enough, this was very speculative. However I do think that the charge up would be a good feature to add while keeping the beam as is now. A short charge up would be akin to needing to lead an opponent rather than the click and hold it is currently. Plus charge up animations are always cool. The rest is more my thoughts without knowing what the meat will be
